datacoves/dbt-coves
# feat - Optionally upload manifest to dbt api

This PR adds optional functionality for uploading the `manifest.json` to the dbt-api pod through an HTTP request. Whether or not the upload happens depends on the environment variable `DATACOVES__UPLOAD_MANIFEST`.

If the upload is enabled, the uploader expects two other custom env variables which are set by Datacoves as part of this PR here: https://github.com/datacoves/datacoves/pull/413

To Test:

Start Airflow and set the `Git Branch` to `testing-manifest-upload2` which points to this branch inside balboa: https://github.com/datacoves/balboa/tree/testing-manifest-upload2

We're currently blocking any HTTP calls between namespaces, so you need to first delete all network policies that we have in place. You can list them with:

```
kubectl get networkpolicies --all-namespaces
```

Then you can delete them with 

```
kubectl delete networkpolicies -n SET_NAMESPACE LIST_ALL_NETWORKPOLICIES
FROM_PREVIOUS_COMMAND_BY_NAME
```

Then run the `yaml_dbt_dag.py` DAG. You'll see that after the `dbt`-command suceeds, `dbt-coves` will upload the manifest to the `dbt-api` pod.

You can see the manifest by fowarding the local minio to localhost with:

```
kubectl port-forward -n core svc/minio 9000 9001
```

Then, go to [localhost:9000](http://localhost:9000) and log in with the credentials found in the secrets `.env`-file of dbt-api.
